src/Compilers/CSharp/Test/Syntax/Parsing/LambdaParameterParsingTests.cs

```diff
@@ -5827,6 +5858,820 @@ public void MissingParameterName_03()
+
+            N(SyntaxKind.ParenthesizedLambdaExpression);
+            {
+                N(SyntaxKind.ParameterList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.Parameter);
+                    {
+                        N(SyntaxKind.ScopedKeyword);
+                        N(SyntaxKind.RefKeyword);
+                        N(SyntaxKind.ReadOnlyKeyword);
+                        N(SyntaxKind.IdentifierToken, "a");
+                    }
+                    N(SyntaxKind.CloseParenToken);
+                }
+                N(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.Block);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestParameterModifierNoType8_a()
+        {
+            string source = "(a, scoped ref readonly b) => { }";
+            UsingExpression(source);
+
+            N(SyntaxKind.ParenthesizedLambdaExpression);
+            {
+                N(SyntaxKind.ParameterList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.Parameter);
+                    {
+                        N(SyntaxKind.IdentifierToken, "a");
+                    }
+                    N(SyntaxKind.CommaToken);
+                    N(SyntaxKind.Parameter);
+                    {
+                        N(SyntaxKind.ScopedKeyword);
+                        N(SyntaxKind.RefKeyword);
+                        N(SyntaxKind.ReadOnlyKeyword);
+                        N(SyntaxKind.IdentifierToken, "b");
+                    }
+                    N(SyntaxKind.CloseParenToken);
+                }
+                N(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.Block);
+                {
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(LanguageVersion.CSharp13)]
+        [InlineData(LanguageVersion.Preview)]
+        public void TestParameterModifierNoType9(LanguageVersion languageVersion)
+        {
+            // 'scoped' is always a type here as the `= null` is not legal in an implicitly typed lambda.
+            string source = "(scoped a = null) => { }";
+            UsingExpression(source, TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+            N(SyntaxKind.ParenthesizedLambdaExpression);
+            {
+                N(SyntaxKind.ParameterList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.Parameter);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "scoped");
```

Comment:
>N(SyntaxKind.IdentifierToken, "scoped"); [](http://example.com/codeflow?start=28&length=40)

This seems inconsistent with `(ref a = 1) => { }` where `ref` is considered a modifier - see `TestParameterModifierNoType6`.

Reply:
Yes.  that's intentional.  Since this has a `= 1` it cannot be an implicitly typed parameter.  As such, it is now an explicitly typed one.  So 'scoped' is the type name.

Reply:
Note: we could unilaterally treat 'scoped' as a modifier.  I just didn't see the need here.  I'll defer to you if you want this.